docs/developer_guide/contributing.rst

```diff
@@ -0,0 +1,420 @@
+Contributing to Plankton
```

Comment:
Not sure if this PR is the place to do it, but I noticed this document has grown very long.

Might be good to split it up a bit, especially since it really shows when viewed in the RTD theme. On the other hand, it automatically generates subsections with anchor links... so maybe that's alright. It is noticeable that you have to scroll a lot more on this page than on others though.

Also, as we discussed before, this is more of a "How To Write Your Own Devices" document than a "Contributing" document. But that could be handled later, when we add real Contribution Guidelines.